be/src/olap/tablet_manager.cpp

```diff
@@ -737,18 +737,16 @@ TabletSharedPtr TabletManager::find_best_tablet_to_compaction(
             }
 
             if (compaction_type == CompactionType::CUMULATIVE_COMPACTION) {
-                if (!table_ptr->try_cumulative_lock()) {
+                MutexLock lock(table_ptr->get_cumulative_lock());
+                if (!lock.own_lock()) {
                     continue;
-                } else {
-                    table_ptr->release_cumulative_lock();
                 }
             }
 
             if (compaction_type == CompactionType::BASE_COMPACTION) {
-                if (!table_ptr->try_base_compaction_lock()) {
+                MutexLock lock(table_ptr->get_base_lock());
+                if (!lock.own_lock()) {
```

Comment:
This is not a try-lock, what does this check mean?

Reply:
I missed it. I will add it in next commit.